spec/mempool/cache.md

```diff
@@ -0,0 +1,36 @@
+# Mempool cache
+
+The *cache* stores the hashes of recently seen transactions. Its purpose is to avoid processing and
+adding duplicate transactions to the mempool.
+
+## Adding transactions to the cache
+
+A node that receives a transaction, before trying to add it to the mempool, puts it in the cache, if
+it is not already there. Only if it is not in the cache, the node will attempt to add it to the
+mempool. If the transaction exists in the cache, it is discarded with an error message.
+
+Another way a transaction may enter the cache is when the node's execution module updates the
```

Comment:
This is another use we have from the cache.

We want to avoid keeping in the mempool transactions that were already committed. Because the mempool is a list of _pending_ or _oustanding_ transactions.

This should be clear also in the introduction.

Reply:
Again, to rephrase in a simpler way:

1. We want to make sure that we don't process transactions that were already committed
2. So we keep on the cache (hashes) of recently committed transactions
3. Upon the update call, if some committed and valid transaction is not yet present in the cache, we add it to the cache.